lib/typesafe.h

```diff
@@ -256,6 +288,11 @@ macro_pure size_t prefix ## _count(const struct prefix##_head *h)              \
 {                                                                              \
 	return h->dh.count;                                                    \
 }                                                                              \
+macro_inline bool prefix ## _is_empty(const struct prefix##_head *h)           \
```

Comment:
Do we really need `*_is_empty()`?  `*_count()` is available everywhere (and not slow either, the lists keep a counter.)

(I'm also picking my brain on whether we can do better with `const`. I think back when I designed this my takeaway was that it's just not necessary, but I'm flexible on that.)

Reply:
just seemed more clear to have the `_is_empty()` option, at least that was the case in a couple of places when I was converting.

yes, const - the issue I encounter most is that a lib has no const, and then later code changes cannot use const because of the lib code. so I do try to notice and add const from time to time.